status/status.go

```diff
@@ -237,6 +243,25 @@ func (s *Status) parseStatusResponse(deployment, podName string, r *models.Statu
 	}
 }
 
+func (s *Status) parseEndpointsResponse(deployment, podName string, eps []*models.Endpoint, err error) {
+	if err != nil {
+		s.AddAggregatedError(deployment, podName, fmt.Errorf("unable to retrieve cilium endpoint information: %s", err))
+		return
+	}
+
+	var notReady uint
+	for _, ep := range eps {
+		if ep != nil && ep.Status != nil && ep.Status.State != nil &&
+			*ep.Status.State != models.EndpointStateReady {
```

Comment:
Can we really capture the endpoint regeneration case in this way? Is it possible that we trigger regeneration and perform this check too early before the regeneration happens? In that case, the state of the endpoint still might be `ready`?

Reply:
AFAICT, restored endpoints are synchronously restored (and marked as restored) before the API is actually exposed, so it should be guaranteed not to be racy. I'm not an expert of that logic though.

-> [d.fetchOldEndpoints](https://github.com/cilium/cilium/blob/7f4e472cc131f5a687ec44524ff106a80cadff3f/daemon/cmd/daemon.go#L652)
---> [endpoint.ReadEPsFromDirNames](https://github.com/cilium/cilium/blob/7f4e472cc131f5a687ec44524ff106a80cadff3f/daemon/cmd/state.go#L171)
-----> [parseEndpoint](https://github.com/cilium/cilium/blob/7f4e472cc131f5a687ec44524ff106a80cadff3f/pkg/endpoint/restore.go#L134)
-------> [ep.setState(StateRestoring, "Endpoint restoring")](https://github.com/cilium/cilium/blob/7f4e472cc131f5a687ec44524ff106a80cadff3f/pkg/endpoint/endpoint.go#L969)

Reply:
Makes sense. Thanks for the clarification. I agree that in this code path, it shouldn't be racy.